### ISSUE TITLE
Add context in UI when a PR job is not executed

### ISSUE THREAD

**Issue by jithin1987** [screwdriver-cd/screwdriver#1045]
## Context

UI lists PR jobs which do not run in 2 cases

1. When using `sourcePaths` none of the changed files in the PR matched a job.
2. PR which was already created gets listed when user does sync from `Options`

In both scenarios we need to give proper explanation in UI on why the build did not ran.

## Objective

1. Add a message in UI explaining why the PR job did not run a build.
1. Message to be exposed when user hovers over the PR job in the PR jobs section in the Pipeline detail page.

**Comment by tkyi** [comment]
Maybe we should work on not creating a PR that will not be run instead of having to explain to users why it was not run.

**Comment by jithin1987** [comment]
> Maybe we should work on not creating a PR that will not be run instead of having to explain to users why it was not run.

Yes, that makes perfect sense. But there can also scenarios where a PR build was not executed. For example when user does `Sync` to fix situations like PRs not showing up and suddenly we display the PR jobs without a build.
